### PR TITLE
Improved the highlighting of Dust syntaxes and helpers a little.

### PR DESCRIPTION
**Pull request by kumarharsh**
Partially fixes Issues #1 and #2, although needs more work.
Also, only the XML version is changed, I've not touched the JSON version.

### CODE REVIEW

File: Syntaxes/DustJS.tmLanguage [hunk 1]
```diff
@@ -30,15 +30,42 @@
...
 		<dict>
 			<key>begin</key>
-			<string>\{[#.&lt;&gt;^/?:]?</string>
+			<string>(\{[@#.&lt;&gt;^/?:]\w+)(\.\w+)?</string>
+			<key>captures</key>
+			<dict>
+				<key>1</key>
+				<dict>
+					<key>name</key>
+					<string>keyword.operator.index-start.php</string>
```
Comment (sntran) [inline Syntaxes/DustJS.tmLanguage]: Usually I would try to follow [TextMate's scope naming convention](http://manual.macromates.com/en/language_grammars). `keyword.operator` is fine, but the last past is usually used for the type of grammar, so in this case, it should be `keyword.operator.dust`.

File: Syntaxes/DustJS.tmLanguage [hunk 2]
```diff
@@ -30,15 +30,42 @@
...
+				</dict>
+				<key>2</key>
+				<dict>
+					<key>name</key>
+					<string>meta.function.js</string>
```
Comment (sntran) [inline Syntaxes/DustJS.tmLanguage]: Same here. However, `meta` is usually used for a bigger scope, mostly the whole line or so, but in here it's only part of the line, so I would suggest `entity.name.function.dust`.

File: Syntaxes/DustJS.tmLanguage [hunk 3]
```diff
@@ -30,15 +30,42 @@
...
+				</dict>
+			</dict>
+			<key>contentName</key>
+			<string>variable.parameter.dust</string>
+			<key>end</key>
+			<string>\/?\}</string>
+			<key>name</key>
+			<string>keyword.operator.index-end.php</string>
```
Comment (sntran) [inline Syntaxes/DustJS.tmLanguage]: Please change from `keyword.operator.index-end.php` to `keyword.operator.dust`!

File: Syntaxes/DustJS.tmLanguage [hunk 4]
```diff
@@ -46,7 +73,7 @@
 			<key>end</key>
 			<string>\/?\}</string>
 			<key>name</key>
-			<string>source.dust.embedded.html</string>
+			<string>entity.name.tag</string>
```
Comment (sntran) [inline Syntaxes/DustJS.tmLanguage]: This part is to handle the rest of the HTML when dust template is embedded into HTML layout. It should be left as `source.dust.embedded.html`.

**Comment by kumarharsh** [comment]
I agree with your points, but I "changed" the scope namings only for changing the colours, because the original colours  clash way too much with the HTML code, and make the dust template less readable.
(I'm using the Solarized theme for testing this, which does not have any support for dust at all... maybe it should be added there too...)

**Comment by sntran** [comment]
I understand that changing scope name is to change the colors, but you need to understand how scope selector works. It will match from the beginning of the scope name, so for example `keyword.operator.dust` will match the `keyword.operator` already, so the last part `.dust` is just for clarification that this belongs to a syntax definition of a dust file.

Quote from the aforementioned TextMate grammar document:

> 1. A minimal theme will only assign styles to 10 of the 11 root groups below (meta does not get a visual style), so you should “spread out” your naming i.e. instead of putting everything below keyword (as your formal language definition may insist) you should think “would I want these two elements styled differently?” and if so, they should probably be put into different root groups.
> 2. Even though you should “spread out” your names, when you have found the group in which you want to place your element (e.g. storage) you should re-use the existing names used below that group (for storage that is modifier or type) rather than make up a new sub-type. You should however append as much information to the sub-type you choose. For example if you are matching the static storage modifier, then instead of just naming it storage.modifier use storage.modifier.static.«language». A scope selector of just storage.modifier will match both, but having the extra information in the name means it is possible to specifically target it disregarding the other storage modifiers.
> 3. Put the language name last in the name. This may seem redundant, since you can generally use a scope selector of: source.«language» storage.modifier, but when embedding languages, this is not always possible.

Also, the themes will never take care of any specific language. They only style the scopes. Depending on the theme, it may not have support for `keyword.operator`, and thus it may not color that scope. However, like I mentioned above, `keyword.operator` and `keyword.operator.dust` will make no difference in coloring. `keyword.operator.php` and `keyword.operator.dust` will also make no difference, but `keyword.operator.dust` makes much more sense in a Dust template highlighting, don't you think?

**Comment by kumarharsh** [comment]
Yup. I agree... I'll redact the request... Lets think about better combos.

**Pull request closed without merging**